File: PYTHON/CH-03/ch03_ex1.py

from pymongo import MongoClient
import os
from dotenv import load_dotenv

load_dotenv()

# MongoDB connection string
uri = os.environ["MONGODB_URI"]

# Connect to MongoDB Atlas
def connect_to_atlas():
    client = MongoClient(uri)
    try:
        print("Connected to MongoDB Atlas")
    # Perform database operations here
    except Exception as e:
        print("Error connecting to MongoDB Atlas:", e)
    finally:
        client.close()
        print("Disconnected from MongoDB Atlas")

connect_to_atlas()

File: JAVASCRIPT/CH-03/ch03_ex1.js

import { MongoClient } from 'mongodb';
require('dotenv').config(); // Load environment variables from .env file

// Get the MongoDB URI from environment variables
const uri = process.env.MONGO_URI;

// Create a new MongoClient
const client = new MongoClient(uri, { useNewUrlParser: true, useUnifiedTopology: true });

async function connectToMongoDB() {
    try {
        await client.connect();
        console.log('Connected to MongoDB');
        // Additional database operations here
    } catch (error) {
        console.error('Error connecting to MongoDB:', error);
    } finally {
        await client.close();
        console.log('Disconnected from MongoDB');
    }
}

connectToMongoDB();
